frontend/src/app/oe/components/blockspans-home/blockspans-home.component.ts

```diff
@@ -144,37 +144,25 @@ export class BlockspansHomeComponent implements OnInit, OnDestroy {
     const blockNumbers = [];
     let blockSpanList = [];
     try {
-      blockSpanList = await lastValueFrom(this.opEnergyApiService.$getBlockSpanList(tipBlock - (span* numberOfSpan) , span, numberOfSpan), {defaultValue: []});
+      blockSpanList = await lastValueFrom(this.opEnergyApiService.$getBlockSpanList(tipBlock, span, numberOfSpan), {defaultValue: []});
     } catch (error) {
       this.toastr.error('Cannot fetch block height data!', 'Failed!');
     }
     blockSpanList.reverse().forEach((blockSpan: BlockSpan) => {
-      blockNumbers.push(blockSpan.endBlockHeight, blockSpan.startBlockHeight);
+      blockNumbers.push(blockSpan.endBlock, blockSpan.startBlock);
     });
     this.pastBlocks = [];
-    forkJoin(
-      blockNumbers.map(
-        blockNo => this.electrsApiService.getBlockHashFromHeight$(blockNo).pipe(
-          switchMap(hash => this.opEnergyApiService.$getBlock(hash))
-        )
-      )
-    )
-    .pipe(take(1))
-    .subscribe((blocks: any[]) => {
-      this.pastBlocks = blocks;
-      this.cd.markForCheck();
-      this.lastPastBlock = this.pastBlocks[0];
-      this.lastPastBlock = {
-        ...this.lastPastBlock,
-        height: this.lastPastBlock.height + this.span
-      };
-      this.location.replaceState(
-        this.router.createUrlTree([(this.network ? '/' + this.network : '') + `/hashstrikes/blockspans/`, this.span, tipBlock]).toString()
-      );
-      this.getTimeStrikes();
-    }, error => {
-      this.toastr.error('Blockspans are not found!', 'Failed!');
-    });
+    this.pastBlocks = blockNumbers;
```

Comment:
it seems, that line 154 should be deleted, right?

Reply:
Yes, we could remove it